### PR TITLE
Refactoring teachers

### PR DESCRIPTION
**Pull request by Titogelo**
This pull request includes a refactoring of teachers, students, classrooms, and dashboard. I have created containers in the redux folder that handle all the status of the different views. The views now split into three different basic views (Form, View, and List). 

The actions have been also slightly modified as some of them use Promises to efficiently handle the requests.

### CODE REVIEW

**Comment by Titogelo** [comment]
I'll refactor actions again make sure they follow the pattern we agreed upon ;-)

**Comment by Titogelo** [comment]
thinking about doing the whole refactoring of actions and reducers in this pull request. It will be much cleaner.

File: src/redux/actions/ClassroomActions.js [hunk 1]
```diff
@@ -4,38 +4,115 @@ import * as types from './ActionTypes';
...
   };
 }
 
+export function fetchClassroom(uid) {
+  return dispatch => {
+    dispatch({
+      type: types.FETCH_CLASSROOM_PENDING,
+      payload: true
+    });
+    new Promise((resolve, reject) => {
```
Comment (miguelcg) [inline src/redux/actions/ClassroomActions.js]: I think this promise never gets resolved

File: src/redux/actions/ClassroomActions.js [hunk 2]
```diff
@@ -4,38 +4,115 @@ import * as types from './ActionTypes';
...
-    payload: database.ref('classrooms/').push({ ...classroom }),
+
+  classroom.created_at = new Date().getTime()/1000;
+
+  return dispatch => {
+    dispatch({
+      type: types.CREATE_CLASSROOM_PENDING,
+      payload: true
+    });
+    new Promise((resolve, reject) => {
```
Comment (miguelcg) [inline src/redux/actions/ClassroomActions.js]: +1

File: src/redux/actions/StudentActions.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+
+export function fetchStudent(uid) {
+  return dispatch => {
+    dispatch({
+      type: types.FETCH_STUDENT_PENDING,
+      payload: true
+    });
+    new Promise((resolve, reject) => {
```
Comment (miguelcg) [inline src/redux/actions/StudentActions.js]: +1

File: src/redux/actions/StudentActions.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+export function createStudent(student) {
+
+  student.created_at = new Date().getTime()/1000;
+
+  return dispatch => {
+    dispatch({
+      type: types.CREATE_STUDENT_PENDING,
+      payload: true
+    });
+    new Promise((resolve, reject) => {
```
Comment (miguelcg) [inline src/redux/actions/StudentActions.js]: +1

File: src/redux/actions/StudentActions.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+export function deleteStudent(uid) {
+
+  return dispatch => {
+    dispatch({
+      type: types.DELETE_STUDENT_PENDING,
+      payload: true
+    });
+    new Promise((resolve, reject) => {
```
Comment (miguelcg) [inline src/redux/actions/StudentActions.js]: +1

**Pull request merged**